### PR TITLE
Detect icmpv6hdr v6

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3323 

Describe changes:
- Adds `icmpv6.hdr` signature keyword
- Fixes typos in comments
- Adds missing doc for `ipv6.hdr` keyword
- Adds a generic `DetectU32Data` structure for signature matches on uint32 values
- Adds `icmpv6.mtu` signature  keyword
- Fix endianness for ICMPv6 MTU decoding

This fixes only the first evasion of the redmine ticket

Makes https://github.com/OISF/suricata-verify/pull/187 pass

Modifies #4632 by taking comments into account

There is still some points to improve/discuss :
- Handle `DetectBufferTypeRegister` as fatal error... Should it be part of this PR ?
- Use `pcre_copy_substring` in detect-tcpmss.c... Should it be part of this PR ?

### CODE REVIEW

**Comment by catenacyber** [comment]
This is rebased on latest master without commits merged in #4637

**Comment by victorjulien** [comment]
Hmm the OSX failure looks suspiciously like the sip failures that are supposed to be fixed.

File: src/detect-engine-prefilter-common.h [hunk 1]
```diff
@@ -19,10 +19,10 @@
 #define __DETECT_ENGINE_PREFILTER_COMMON_H__
 
 typedef union {
-    uint8_t u8[8];
-    uint16_t u16[4];
-    uint32_t u32[2];
-    uint64_t u64;
+    uint8_t u8[16];
```
Comment (victorjulien) [inline src/detect-engine-prefilter-common.h]: this change belongs in it's own commit

File: src/detect-engine-prefilter-common.h [hunk 2]
```diff
@@ -94,4 +94,14 @@ PrefilterPacketHeaderExtraMatch(const PrefilterPacketHeaderCtx *ctx,
     return TRUE;
 }
 
+static inline bool PrefilterIsPrefilterableById(const Signature *s, enum DetectKeywordId kid) {
+    const SigMatch *sm;
+    for (sm = s->init_data->smlists[DETECT_SM_LIST_MATCH] ; sm != NULL; sm = sm->next) {
+        if (sm->type == kid) {
+            return TRUE;
```
Comment (victorjulien) [inline src/detect-engine-prefilter-common.h]: `true`

File: src/detect-engine-prefilter-common.h [hunk 3]
```diff
@@ -94,4 +94,14 @@ PrefilterPacketHeaderExtraMatch(const PrefilterPacketHeaderCtx *ctx,
...
+        }
+    }
+    return FALSE;
```
Comment (victorjulien) [inline src/detect-engine-prefilter-common.h]: `false`

File: src/detect-engine-prefilter-common.h (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-prefilter-common.h]: ok

File: src/detect-engine-prefilter-common.h (hunk 2 above)
Comment (catenacyber) [reply src/detect-engine-prefilter-common.h]: indeed, my bad revert

File: src/detect-engine-prefilter-common.h (hunk 3 above)
Comment (catenacyber) [reply src/detect-engine-prefilter-common.h]: indeed, my bad revert
